api/types/database.go

```diff
@@ -405,19 +430,74 @@ func (d *DatabaseV3) CheckAndSetDefaults() error {
 	return nil
 }
 
-// parseRDSEndpoint extracts region from the provided RDS endpoint.
-func parseRDSEndpoint(endpoint string) (instanceID, region string, err error) {
+// rdsEndpointDetails contains information about an RDS endpoint.
+type rdsEndpointDetails struct {
+	instanceID        string
+	clusterID         string
+	proxyName         string
+	proxyEndpointName string
+	region            string
+}
+
+// parseRDSEndpoint extracts identifiers and region from the provided RDS endpoint.
+func parseRDSEndpoint(endpoint string) (*rdsEndpointDetails, error) {
 	host, _, err := net.SplitHostPort(endpoint)
 	if err != nil {
-		return "", "", trace.Wrap(err)
+		return nil, trace.Wrap(err)
 	}
-	// RDS/Aurora endpoint looks like this:
+
+	// RDS/Aurora instance endpoints look like this:
 	// aurora-instance-1.abcdefghijklmnop.us-west-1.rds.amazonaws.com
+	//
+	// Aurora cluster endpoints look like this:
+	// my-cluster.cluster-abcdefghijklmnop.us-west-1.rds.amazonaws.com
+	// my-cluster.cluster-ro-abcdefghijklmnop.us-west-1.rds.amazonaws.com
+	// my-custom.cluster-custom-abcdefghijklmnop.us-west-1.rds.amazonaws.com
+	//
+	// RDS proxy "default" endpoints look like this:
+	// my-proxy.proxy-abcdefghijklmnop.us-west-1.rds.amazonaws.com
+	//
+	// Additional "proxy endpoints" can be added to a RDS proxy and look like:
+	// my-proxy-custom.endpoint.proxy-abcdefghijklmnop.us-west-1.rds.amazonaws.com
+	//
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Overview.Endpoints.html
+	// https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/rds-proxy-setup.html#rds-proxy-connecting
+	// https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/rds-proxy-endpoints.html
+	if !strings.HasSuffix(host, rdsEndpointSuffix) {
+		return nil, trace.BadParameter("endpoint %v does not contain RDS endpoint suffix %v", endpoint, rdsEndpointSuffix)
+	}
+
 	parts := strings.Split(host, ".")
-	if !strings.HasSuffix(host, rdsEndpointSuffix) || len(parts) != 6 {
-		return "", "", trace.BadParameter("failed to parse %v as RDS endpoint", endpoint)
+
+	// The RDS proxy endpoints have one extra level of subdomains.
+	if len(parts) == 7 && strings.HasPrefix(parts[2], rdsProxySubdomainPrefix) {
```

Comment:
All this will probably have to be refactored with the other change we did to support Chinese endpoints.